example-apps/chatbot-rag-app/k8s-manifest.yml

```diff
@@ -0,0 +1,55 @@
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: chatbot-rag-app
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: chatbot-rag-app
+  template:
+    metadata:
+      labels:
+        app: chatbot-rag-app
+    spec:
+      # For `LLM_TYPE=vertex`: create a volume for application_default_credentials.json
+      volumes:
+        - name: gcloud-credentials
+          secret:
+            secretName: gcloud-credentials
```

Comment:
Sorry realized one thing post approve - how about populating the secret in this file with a dummy value? I think currently the deployment will fail if the step to create the secret manually isn't done.

Reply:
lemme go from scratch not creating the secret. if true I'll try to figure out how to do what you've mentioned as indeed we don't want a failure on something optional and for many obscure.

Reply:
totally right..

```
Events:
  Type     Reason       Age               From               Message
  ----     ------       ----              ----               -------
  Normal   Scheduled    39s               default-scheduler  Successfully assigned default/chatbot-rag-app-7984995bc8-pwqxg to colima
  Warning  FailedMount  8s (x7 over 40s)  kubelet            MountVolume.SetUp failed for volume "gcloud-credentials" : secret "gcloud-credentials" not found
```

Reply:
got that working, then ran it by grok 3 think mode which told me all I need to do is mark this optional ;)